Assets/Simon/Scripts/Sequence.cs

```diff
@@ -0,0 +1,55 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System;
+
+public class Sequence : MonoBehaviour
+{
+    public List<Colors> ColorSequence { get; private set; }
+
+    private int sequenceIndex = 0;
+    public void CreateColorSequence(int sequenceLength)
+    {
+        if(sequenceLength <= 0)
+        {
+            Debug.LogError("The length of the sequence should be larger than 0");
+        }
+
+        ColorSequence = new List<Colors>();
+
+        for(int i = 0; i < sequenceLength; i++)

```

Comment:
Instead of making a new List then using a for loop, you could use LINQ (functional paradigm).

```csharp

        ColorSequence = Enumerable.Range(0, sequenceLength - 1)
            .Select(x => getRandomColor()).ToList();
```

This might be slower than a for loop. I just like fluent interfaces (method chaining) lol.
https://en.wikipedia.org/wiki/Fluent_interface

Reply:
I don't know anything about LINK yet. Is it okay if we keep things as they are now? I don't want to just copy paste something I don't understand.

Reply:
Yeah leave it as is. I just put it there if you were curious.

Reply:
Also, never just copy paste code :D Should always test it and makes sure it works in the context of your app.